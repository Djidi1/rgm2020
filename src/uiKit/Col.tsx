import styled from 'styled-components';

interface ColProps {
  align?: string;
  width?: string;
  margin?: string;
}

const Col = styled('div')<ColProps>`
  display: flex;
  flex: 1 0 ${({ width }) => width || '100%'};
  flex-flow: column nowrap;
  max-width: 100%;
  position: relative;
  min-height: 1px;
  margin: ${({ margin }) => margin || 'initial'};
  text-align: ${({ align }) => align || 'left'};
`;

export default Col;
