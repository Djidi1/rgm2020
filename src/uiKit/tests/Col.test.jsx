import React from 'react';
import renderer from 'react-test-renderer';
import Col from '../Col';

describe('UI-Kit Col', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('is expected to render successfully', () => {
    const wrapper = renderer
      .create(<Col />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
