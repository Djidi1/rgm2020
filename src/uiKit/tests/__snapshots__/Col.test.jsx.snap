// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI-Kit Col is expected to render successfully 1`] = `
<div
  className="Col-sc-1dskdwa-0 kVzFTl"
/>
`;
